core/infrastructure/kafka/v1beta1/kafka_types.go

```diff
@@ -53,20 +53,19 @@ func (kafkaMap *KafkaMap) DeepCopy() *KafkaMap {
 
 // KafkaClusterSpec defines the desired state of Kafka Cluster
 type KafkaClusterSpec struct {
-	Replicas   int32          `json:"replicas,omitempty"`
-	Listeners  KafkaListeners `json:"listeners,omitempty"`
-	Storage    KafkaStorage   `json:"storage,omitempty"`
-	Config     KafkaMap       `json:"config,omitempty"`
-	JvmOptions KafkaMap       `json:"jvmOptions,omitempty"`
-}
-
-// KafkaListeners Configures the broker authorization
-type KafkaListeners struct {
-	Plain KafkaListenerPlain `json:"plain,omitempty"`
-}
-
-// KafkaListenerPlain Listener type Plain
-type KafkaListenerPlain struct {
+	Replicas   int32                  `json:"replicas,omitempty"`
+	Listeners  []GenericKafkaListener `json:"listeners,omitempty"`
+	Storage    KafkaStorage           `json:"storage,omitempty"`
+	Config     KafkaMap               `json:"config,omitempty"`
+	JvmOptions KafkaMap               `json:"jvmOptions,omitempty"`
+}
+
+// GenericKafkaListener ...
+type GenericKafkaListener struct {
```

Comment:
Why not just `KafkaListener`?

Reply:
We could use `KafkaListener` as well. I used same name as used by Strimzi-operator developers to refer Listener.
Previously they have java class with name `KafkaListener`, but now they had mark it deprecated and create new class with name `GenericKafkaListener`